eldritch/test_events.py

```diff
@@ -120,6 +148,123 @@ def testMoveMultipleSpaces(self):
     self.assertEqual(self.char.place.name, "Graveyard")
     self.assertEqual(self.char.movement_points, 1)
 
+  def testIllegalMoveMultipleSpaces(self):
+    self.char.movement_points = 1
+    movement = Movement(
+        self.char, [self.state.places[name] for name in ["Easttown", "Rivertown", "Graveyard"]])
+    self.assertFalse(movement.is_resolved())
+    self.assertEqual(self.char.movement_points, 1)
+
+    self.state.event_stack.append(movement)
+    self.resolve_until_done()
+
+    self.assertTrue(movement.is_resolved())
+    self.assertEqual(self.char.place.name, "Easttown")
+    self.assertEqual(self.char.movement_points, 0)
+
+  def testMoveMultipleThroughMonster(self):
+    self.char.fight_will_slider = 2
+    cultist = next(monster for monster in self.state.monsters if monster.name == "Cultist")
+    cultist.place = self.state.places["Easttown"]
+    movement = Movement(
+        self.char, [self.state.places[name] for name in ["Easttown", "Rivertown", "Graveyard"]])
+    self.assertFalse(movement.is_resolved())
+    self.assertEqual(self.char.movement_points, 4)
+
+    self.state.event_stack.append(Sequence([movement, EndMovement(self.char)], self.char))
+    choice = self.resolve_to_choice(MultipleChoice)
+    self.assertEqual(choice.choices, ["Fight", "Evade"])
+    choice.resolve(self.state, "Fight")
+
+    next_choice = self.resolve_to_choice(MultipleChoice)
+    self.assertEqual(next_choice.choices, ["Fight", "Flee"])
+    next_choice.resolve(self.state, "Fight")
+
+    third_choice = self.resolve_to_choice(CombatChoice)
+    self.assertIsNone(third_choice.choices)
+    third_choice.resolve(self.state, [])
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=5)):
+      self.resolve_until_done()
+
+    self.assertTrue(movement.is_resolved())
+    self.assertEqual(self.char.place.name, "Easttown")
+    # self.assertEqual(self.char.movement_points, 0)
+    # TODO: ^ should this pass?
+    # self.assertIn(cultist, self.char.possessions)
+    # TODO: take the monster as a trophy
+
+  def testMoveMultipleThroughMonsterFailedEvade(self):
+    self.char.fight_will_slider = 2
+    self.char.speed_sneak_slider = 0
```

Comment:
You're setting the speed_sneak_slider to 0, but then asserting that the character has 4 movement points. A bit weird, though I suppose it works if you need the extra sneak to make the test work.

Reply:
Nah, that was mostly me trying to make changes to it, then finding that it did nothing because of your observation below that we weren't actually in the movement phase.